gitops-engine/pkg/diff/diff.go

```diff
@@ -286,7 +288,63 @@ func removeWebhookMutation(predictedLive, live *unstructured.Unstructured, gvkPa
 	if !ok {
 		return nil, fmt.Errorf("error converting live typedValue: expected map got %T", plu)
 	}
-	return &unstructured.Unstructured{Object: pl}, nil
+	resultPredictedLive := &unstructured.Unstructured{Object: pl}
+
+	// Additional fix for issue #24882: remove keys from ConfigMap/Secret data fields
+	// that are not present in config. This handles the case where server-side apply
+	// doesn't remove keys due to field ownership conflicts.
+	resultPredictedLive = removeExtraMapKeys(config, resultPredictedLive, gvk)
+
+	return resultPredictedLive, nil
+}
+
+// removeExtraMapKeys removes extra keys from ConfigMap/Secret data fields that are not present in config.
+// This handles the case where server-side apply doesn't remove keys due to field ownership conflicts.
+func removeExtraMapKeys(config, predictedLive *unstructured.Unstructured, gvk schema.GroupVersionKind) *unstructured.Unstructured {
+	// Only apply this fix for ConfigMaps and Secrets
+	if gvk.Group == "" && (gvk.Kind == "ConfigMap" || gvk.Kind == "Secret") {
```

Comment:
When it comes to ServerSide Diff, we don't want to make any specific kind checks in the codebase. Our goal is to mimic exactly what Kubernetes does when we sync with server-side apply enabled. This behaviour should be driven by the `.metadata.manageFields` and the resource schema that lives in k8s.